packages/firestore/test/unit/local/simple_db.test.ts

```diff
@@ -142,18 +149,13 @@ describe('SimpleDb', () => {
   });
 
   it('aborts transactions when an error happens', async () => {
-    let gotError = false;
-    try {
-      await runTransaction(store => {
+    await expect(
+      runTransaction(store => {
         return store.put(dummyUser).next(() => {
-          throw new Error('error');
+          throw new Error('Generated error');
         });
-      });
-    } catch (error) {
-      expect(error.message).to.equal('error');
-      gotError = true;
-    }
-    expect(gotError).to.equal(true);
+      })
+    ).to.eventually.be.rejectedWith('Generated error');
```

Comment:
FWIW, looking at the chai-as-promised examples, they don't tend to use expect() so this would become:

```
await runTransaction(...).should.eventually.be.rejectedWith('Generated error');
```

I don't care strongly though, so feel free to leave as-is.

Reply:
I changed this to use `expect` since I found it somewhat questionable to just extend the Promise implementation. Using `expect` hides this a little bit.